database.py

```diff
@@ -574,6 +557,73 @@ def get_submission_by_id(self, id: SubmissionId, /) -> Optional[Submission]:
 
         return None
 
+    def get_submissions_by_ids(self, ids: tuple[SubmissionId, ...], /) -> list[Submission]:
+        """Retrieve fully-hydrated Submission objects for the specified Submission IDs"""
+        if len(ids) == 0:
+            return []
+        if len(ids) == 1:
+            res = self.get_submission_by_id(ids[0])
+            if res is None:
+                return []
+            return [res]
+
+        # sqlite doesn't let us pass a list of IDs directly. Other DBs do, not sqlite.
+        # TODO: Paginate if hundreds of IDs. For sizes we care about, this is fine.
+        query = """
+            SELECT submission_id, year, day_part, user, average_time, code, valid, submitted_at, bencher_version, benchmark_format 
+            FROM submissions 
+            WHERE submission_id in ({})
+            """
+
+        query = query.format(", ".join(["?"] * len(ids)))
+
+        results = list(self._cursor.execute(query, ids))
+
+        out = []
+
+        for (
+            subm_id,
+            year,
+            day_part,
+            user_id,
+            avg_time,
+            code,
+            valid,
+            submitted_at,
+            bencher_version,
+            benchmark_format,
+        ) in results:
+            (day, part) = unpack_day_part(day_part)
+
+            benches = list[BenchmarkRun](
+                BenchmarkRun(
+                    subm_id, Picoseconds(average_time), label, answer, dt_from_unix(completed_at)
+                )
+                for label, average_time, answer, completed_at in self._cursor.execute(
+                    "SELECT session_label, average_time, answer, completed_at FROM benchmark_runs WHERE (submission = ?)",
+                    (subm_id,),
+                )
```

Comment:
This is an instance of the N+1 Queries Problem, and there's better ways to solve this. But, whatever, it works for now.

Reply:
sqlite lends itself well to N+1 btw, since its a library, not a server